### DISCUSSION TITLE
Testbench Core v7.9 Gets Installed With Laravel 9.2.0 (and cause Access to undeclared static property $latestResponse error)

### DISCUSSION THREAD

**Discussion by fulopattila122** [orchestral/testbench-core#85] [Q&A] [answered]
Hi There,

I'm kinda puzzled by the issue below, which is - as far as I see - likely a composer issue I never met before.

We have packages that support Laravel 9.2 as the minimum version, and we instrument the CI pipeline to install Laravel 9.2.
Orchestral Testbench and Testbench Core have properly set their Laravel dependencies within the 7.x series,
still, for some reason, Testbench Core 7.9 gets installed along with Laravel 9.2.0:

![image](https://user-images.githubusercontent.com/1162360/194509426-a4730224-6b3e-4ee2-9139-ae4ae0a98bb2.png)

What I find odd is that composer properly detects the conflict between Laravel 9.2.0 and Testbench 7.2.0:

![image](https://user-images.githubusercontent.com/1162360/194510941-42a6ae48-00ec-4ace-b0cc-75c1be78b119.png)

..but with Testbench Core, it sees no such constraints:

![image](https://user-images.githubusercontent.com/1162360/194511169-3b858390-2522-4b28-95f5-23f7bf1d7fe0.png)

Even though composer is apparently aware of the minimum Laravel requirement, (which is not met):

![image](https://user-images.githubusercontent.com/1162360/194511575-0e04c2e0-b93c-4165-b531-0eba215891af.png)

What I would expect is that composer detects that Orchestra Testbench Core v7.1 is the highest version it should install and act accordingly.

I tried:
- Testing locally and in CI
- Testing with another package
- Upgrading composer to the latest (2.4.2) version
- Adding `orchestra/testbench-core` as a root requirement (besides `orchestra/testbench`)

all bringing the same result.

Does anyone have an idea what is causing this?

**Comment by crynobone** [comment, marked as answer]
Can't debug without looking at the actual `composer.json` and CI pipeline.

**Reply by fulopattila122** [reply]
Thanks for taking a look at it!
Here's the pipeline: https://github.com/vanilophp/braintree/actions/runs/3202767361
and here's the `composer.json`: https://github.com/vanilophp/braintree/blob/master/composer.json

Another instance/repo is this: https://github.com/vanilophp/framework/actions/runs/3202991945/jobs/5232567471

Thanks again! :bow:

**Reply by crynobone** [reply]
You issue is that you force Laravel Framework to require 9.2.x but everything else can install highest dependencies. When testing lowest deps you should ensure all dependencies are running on the lowest version and not just the Framework.

**Reply by fulopattila122** [reply]
Thanks a lot for the quick solution. It solves the problem in the CI; however, I still don't get how composer allows testbench core 7.9 besides Laravel 9.2 when testbench core clearly states that it requires at least Laravel 9.34.

But this is obviously not a testbench, but a composer issue, since both testbench and testbench-core packages are properly specifying their minimum versions. And this constraint gets applied in the case of testbench but not for testbench-core.

Anyway, thanks a lot for your help! :bowing_man:

**Reply by crynobone** [reply]
Testbench Core cannot have any constraints against Laravel Framework because the Framework itself requiring Testbench Core for Integration Tests.

**Reply by fulopattila122** [reply]
As far as I see it does :thinking: https://github.com/orchestral/testbench-core/blob/v7.9.0/composer.json#L39
And I think it's even very correct, due to the BC in 9.32

**Reply by fulopattila122** [reply]
Noooow I got it!! It's a dev requirement, which means that it doesn't apply in this case :man_facepalming: 
Sorry for bothering 😬
